Framework/API/src/WorkspaceOpOverloads.cpp

```diff
@@ -538,6 +538,11 @@ void WorkspaceHelpers::makeDistribution(MatrixWorkspace_sptr workspace,
                              "into distributions.");
 
   const size_t numberOfSpectra = workspace->getNumberHistograms();
+  // check X vs Y data size
+  if (workspace->x(0).size() == workspace->y(0).size()) {
```

Comment:
You can use `workspace->histogram(0).xMode() == Histogram::XMode::Points` instead of the size check. That is directly stating the intent of the check and you can avoid the code comment above.